src/main/java/io/ticktok/rest/RestTicktokClient.java

```diff
@@ -57,11 +57,11 @@ private void validateResponseWasOk(HttpResponse response) {
         }
     }
 
-    public Clock register(String schedule) throws IOException {
-        return new Gson().fromJson(post(handleBody(schedule), TicktokApi.REGISTER_NEW_CLOCK), Clock.class);
+    public Clock register(String name, String schedule) throws IOException {
```

Comment:
You shouldn't throw Checked exception out of your classes

Reply:
in this case im not sure i have a better option.
we should fail if request is failed and externalize it to the user

Reply:
you can catch it and throw your own non checked exception with is better